modules/runner/src/integrationTest/java/org/apache/ignite/internal/sql/engine/datatypes/varbinary/ItVarBinaryIndexTest.java

```diff
@@ -18,20 +18,131 @@
+    }
+
+    /**
+     * Key lookup. Literal, cast, cast with precision against different column types.
+     */
+    @ParameterizedTest
+    @MethodSource("indexChecks")
+    public void testKeyLookUp2(String table, ValueMode mode) {
+        VarBinary value1 = values.get(0);
+        String value1str = mode.toSql(testTypeSpec, value1);
+
+        // TODO Disable for VARBINARY, remove after https://issues.apache.org/jira/browse/IGNITE-19931 is fixed
+        Assumptions.assumeFalse(mode == ValueMode.CAST);
+
+        checkQuery(format("SELECT * FROM {} WHERE test_key = {}", table, value1str))
+                .matches(containsIndexScan("PUBLIC", table, table + "_TEST_KEY_IDX"))
+                .returns(1, value1)
+                .check();
+    }
+
+    /** {@inheritDoc} */
+    @ParameterizedTest
+    @MethodSource("compoundIndex")
+    @Override
+    public void testCompoundIndex(TestTypeArguments<VarBinary> arguments) throws InterruptedException {
+        // TODO Disable for VARBINARY, remove after https://issues.apache.org/jira/browse/IGNITE-19931 is fixed.
+        // Lookups for VARBINARY and VARCHAR/CHAR work
+        Assumptions.assumeFalse(arguments.toString().startsWith("VARBINARY"));
+
+        super.testCompoundIndex(arguments);
+    }
+
+    /** {@inheritDoc} */
+    @Test
+    @Override
+    @Disabled("https://issues.apache.org/jira/browse/IGNITE-19931")
+    public void testIndexDynParam() {
+        super.testIndexDynParam();
+    }
+
     /** {@inheritDoc} */
     @Override
     protected DataTypeTestSpec<VarBinary> getTypeSpec() {
         return DataTypeTestSpecs.VARBINARY_TYPE;
     }
+
+    private static Stream<Arguments> indexChecks() {
+        return Stream.of(
+                Arguments.of(Named.of("VARBINARY_DEFAULT_LENGTH", "T"), ValueMode.LITERAL),
+                //Arguments.of(Named.of("VARBINARY_DEFAULT_LENGTH", "T"), ValueMode.CAST),
+                // duplicates a test case defined in the base class.
+                Arguments.of(Named.of("VARBINARY_DEFAULT_LENGTH", "T"), ValueMode.CAST_WITH_PRECISION),
+
+                Arguments.of("VARBINARY_FIXED_LENGTH", ValueMode.LITERAL),
+                Arguments.of("VARBINARY_FIXED_LENGTH", ValueMode.CAST),
+                Arguments.of("VARBINARY_FIXED_LENGTH", ValueMode.CAST_WITH_PRECISION),
+
+                Arguments.of("BINARY_FIXED_LENGTH", ValueMode.LITERAL),
+                Arguments.of("BINARY_FIXED_LENGTH", ValueMode.CAST),
+                Arguments.of("BINARY_FIXED_LENGTH", ValueMode.CAST_WITH_PRECISION)
+        );
+    }
+
+    /**
+     * Value generation mode.
+     */
+    public enum ValueMode {
+        LITERAL,
+        CAST,
+        CAST_WITH_PRECISION;
+
+        <T extends Comparable<T>> String toSql(DataTypeTestSpec<T> spec, T value) {
+            switch (this) {
+                case LITERAL:
+                    return spec.toLiteral(value);
+                case CAST: {
+                    String str = spec.toStringValue(value);
```

Comment:
https://issues.apache.org/jira/browse/IGNITE-20185 will reject casting from char to varbinary i.e. '1'::varbinary would be rejected, also it`s not allowed by standard. But i also understand that this is without the scope of this issue but if you can easily fix it - it would be great.

Reply:
Replaced string literals with binary literals.